src/org/traccar/database/NotificationManager.java

```diff
@@ -130,8 +130,11 @@ public Notification getUserNotificationByType(long userId, String type) {
     public void updateNotification(Notification notification) {
         Notification cachedNotification = getUserNotificationByType(notification.getUserId(), notification.getType());
         if (cachedNotification != null) {
-            if (!cachedNotification.getAttributes().equals(notification.getAttributes())) {
-                if (notification.getAttributes().isEmpty()) {
+            if (cachedNotification.getWeb() != notification.getWeb()
+                    || cachedNotification.getMail() != notification.getMail()
+                    || !cachedNotification.getAttributes().equals(notification.getAttributes())) {
```

Comment:
What is this code used for? We might use attributes for something later, but for now we are not using them.